packages/typespec-autorest/test/openapi-output.test.ts

```diff
@@ -847,6 +847,228 @@ describe("typespec-autorest: extension decorator", () => {
   });
 });
 
+describe("typespec-azure: identifiers decorator", () => {
+  it("ignores name/id keys for x-ms-identifiers", async () => {
+    const oapi = await openApiFor(
+      `
+      model Pet {
+        @key
+        name: string;
+        @key
+        id: int32;
+      }
+      model PetList {
+        value: Pet[]
+      }
+      @route("/Pets")
+      @get op list(): PetList;
+      `,
+    );
+    ok(oapi.paths["/Pets"].get);
+    deepStrictEqual(oapi.definitions.PetList.properties.value["x-ms-identifiers"], undefined);
+  });
+  it("uses identifiers decorator for properties", async () => {
+    const oapi = await openApiFor(
+      `
+      @armProviderNamespace
+      @useDependency(Azure.ResourceManager.Versions.v1_0_Preview_1)
+      namespace Microsoft.Test;
+      
+      model Pet {
+        name: string;
+        age: int32;
+      }
+      model PetList {
+        @identifiers(["age"])
+        value: Pet[]
+      }
+      @route("/Pets")
+      @get op list(): PetList;
+      `,
+    );
+    ok(oapi.paths["/Pets"].get);
+    deepStrictEqual(oapi.definitions.PetList.properties.value["x-ms-identifiers"], ["age"]);
+  });
+  it("identifies keys correctly as x-ms-identifiers", async () => {
+    const oapi = await openApiFor(
+      `
+      @armProviderNamespace
+      @useDependency(Azure.ResourceManager.Versions.v1_0_Preview_1)
+      namespace Microsoft.Test;
+      
+      model Pet {
+        name: string;
+        @key
+        age: int32;
+      }
+      model PetList {
+        value: Pet[]
+      }
+      @route("/Pets")
+      @get op list(): PetList;
+      `,
+    );
+    ok(oapi.paths["/Pets"].get);
+    deepStrictEqual(oapi.definitions.PetList.properties.value["x-ms-identifiers"], ["age"]);
+  });
+  it("x-ms-identifiers ignores keys for non armProviderNamespace", async () => {
+    const oapi = await openApiFor(
+      `
+      model Pet {
+        name: string;
+        @key
+        age: int32;
+      }
+      model PetList {
+        value: Pet[]
+      }
+      @route("/Pets")
+      @get op list(): PetList;
+      `,
+    );
+    ok(oapi.paths["/Pets"].get);
+    deepStrictEqual(oapi.definitions.PetList.properties.value["x-ms-identifiers"], []);
```

Comment:
In this case, it should inject no extension at all, not `x-ms-identifiers: []`

Reply:
I think this can be a little bit more impactful as it is the default behavior right now ([playgroun example](https://azure.github.io/typespec-azure/playground/?c=aW1wb3J0ICJAdHlwZXNwZWMvaHR0cCI7CtIZcmVzdNUZdmVyc2lvbmluZ8wfYXp1cmUtdG9vbHMvyCstxhVjb3JlIjsKCnVzaW5nIFR5cGVTcGVjLkh0dHA70BVSZXN00RVWyXLIG0HEWS5Db3Jl0hIuVHJhaXRzOwoKCiAgbW9kZWwgUGV0IHsKICAgIG5hbWU6IHN0csVNxBJhZ2U6IGludDMyxBB9zDRMaXPIOHZhbHVlOsQVW13HJ0Byb3V0ZSgiL1BldHMiKcQSZ2V0IG9wIGxpc3QoKcUuxEM7&e=%40azure-tools%2Ftypespec-autorest&options=%7B%22linterRuleSet%22%3A%7B%22extends%22%3A%5B%22%40azure-tools%2Ftypespec-azure-rulesets%2Fdata-plane%22%5D%7D%7D)) is that ok?

Reply:
Yeah, I didn't realize we were doing this for data plane specs.  If there is no behavior change then there's no reason not to do this